### PR TITLE
Produce an error when loading a mismatching calibration

### PR DESCRIPTION
**Pull request by PeterJCLaw**
If we use a calibration for a resolution other than that which it is intended, we'll get nonsense values. Ensure that that doesn't happen by requiring that `load_camera_calibrations` is passed the image resolution which will be used and then validating that against the value in the calibration file.

### CODE REVIEW

File: tests/test_load_calibrations.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    with mock.patch(
+        'pathlib.Path.open',
+        return_value=(TEST_DATA / 'C016_calibration.xml').open(),
+    ):
+        with pytest.raises(ResolutionMismatchError):
+            load_camera_calibrations('camera-model', (100, 100))
+
+
+def test_accepts_matching_resolution():
+    """Ensure that loading a mismatched calibration file produces an error."""
```
Comment (Adimote) [inline tests/test_load_calibrations.py]: Looks like you forgot to update this docstring

**Review by Adimote — CHANGES_REQUESTED** [review]
Looks excellent as usual, there's an incorrect docstring however.

**Comment by PeterJCLaw** [comment]
Apologies, I should have held fire before PRifying this; rebase incoming.

File: tests/test_load_calibrations.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    with mock.patch(
+        'pathlib.Path.open',
+        return_value=(TEST_DATA / 'C016_calibration.xml').open(),
+    ):
+        with pytest.raises(ResolutionMismatchError):
+            load_camera_calibrations('camera-model', (100, 100))
+
+
+def test_accepts_matching_resolution():
+    """Ensure that loading a matcing calibration file works."""
```
Comment (Adimote) [inline tests/test_load_calibrations.py]: Miss-spelt `matcing`

File: tests/test_load_calibrations.py (hunk 1 above)
Comment (PeterJCLaw) [reply tests/test_load_calibrations.py]: Yup; copy/pasta error: bd5fd55b26efc21b16f84ead192b85589b5cc37c

File: tests/test_load_calibrations.py (hunk 2 above)
Comment (PeterJCLaw) [reply tests/test_load_calibrations.py]: hnnng

Comment (PeterJCLaw) [reply tests/test_load_calibrations.py]: dada343

**Review by Adimote — APPROVED** [review]

**Comment by Adimote** [comment]
:shipit:

**Pull request merged**